File: scripts/convert.jl
# Import the `run` function from the `Base` module
import Base.run

# Define the file paths
docx_file = "sebvc_Resume.docx"
md_file = "sebvc_Resume.md"

# Convert .docx to .md using Pandoc
# include word styles in md with +styles
# github flavored markdown with gfm
# ../Resume> pandoc -f docx+styles -t gfm sebvc_Resume.docx -o sebvc_Resume.md 
run(`pandoc -f docx+styles -t gfm $docx_file -o $md_file`)

# Read the contents of the .md file
md_content = read(md_file, String)

# Define the custom lines to add at the beginning, I am on Windows Hence the use of \r\n to write
custom_lines = """<!-- Julia write() to include .CSS reference -->\r\n<head>\r\n  <link rel="stylesheet" href="media/style_block_insert.css">\r\n</head>\r\n<!--------------------------------------------->\r\n"""

# Prepend the custom lines to the existing content
md_content = custom_lines * md_content

# Write the updated content back to the .md file
write(md_file, md_content)

File: scripts/convert.py
import subprocess

def convert_docx_to_md(docx_file, md_file, custom_lines):
    # Convert the .docx file to .md using Pandoc
	# include word styles in md
	# github flavored markdown
	# ../Resume/> pandoc -f docx+styles -t gfm sebvc_Resume.docx -o sebvc_Resume.md
    pandoc_command = ['pandoc', '-f', 'docx+styles', '-t', 'gfm', docx_file, '-o', md_file]
    subprocess.run(pandoc_command)

    # Read the content of the generated .md file
    with open(md_file, 'r') as file:
        content = file.read()

    # Insert the custom lines at the beginning of the content
    custom_lines = '\n'.join(custom_lines)
    content_with_custom_lines = f"{custom_lines}\n{content}"

    # Write the modified content back to the .md file
    with open(md_file, 'w') as file:
        file.write(content_with_custom_lines)

# Specify the paths and custom lines
docx_file_path = 'sebvc_Resume.docx'
md_file_path = 'sebvc_Resume.md'
custom_lines = [
	'<!-- Python file.write() to include .CSS reference -->',
    '<head>',
    '  <link rel="stylesheet" href="media/style_block_insert.css">',
    '</head>',
	'<!--------------------------------------------------->'
    ]

# Call the conversion function
convert_docx_to_md(docx_file_path, md_file_path, custom_lines)
